### PR TITLE
Document existing code

### LINKED ISSUE #1: Document existing code
Need to give all existing classes and methods their own docstrings to appear in IDE suggestions.

### PR DESCRIPTION
**Pull request by ghost**
Added documentation for .py files. Please review as group before merging with master branch.

### CODE REVIEW

File: api/pokeapi.py [hunk 1]
```diff
@@ -5,20 +5,23 @@
...
 def get_random_pokemon_from_api() -> Pokemon:
+    """Returns a random Pokemon from the PokeAPI"""
+
     rand_dex_id = random.randint(1, MAX_DEX_NUMBER)
     pokepy_mon = client.get_pokemon(rand_dex_id)
     return map_pokepy_to_pokemon(pokepy_mon)
 
 
 def map_pokepy_to_pokemon(pokepy_mon: any) -> Pokemon:
+    """Converts the data pulled from the PokeApi and maps it to the Pokemon class."""
```
Comment (biggestcookie) [inline api/pokeapi.py]: Minor: remove period

File: app.py [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 
 def main():
+    """Entry point for application"""
```
Comment (biggestcookie) [inline app.py]: Minor: `for the application`

File: config/config.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
 import json
 
+"""Holds flavor text"""
```
Comment (biggestcookie) [inline config/config.py]: I think it's more than just flavor text, since it's essentially almost all the text strings fed back to the user

File: game/player.py [hunk 4]
```diff
@@ -5,10 +5,14 @@
 
 
 class Player:
+    """Holds current information about the player"""
+
     team: List[Pokemon] = []
 
     def __init__(self):
-        if (len(self.team) < 1):
+        """fills it with a random Pokemon from PokeApi"""
```
Comment (biggestcookie) [inline game/player.py]: What is `it`? Also minor: capitalization

File: game/pokemon.py [hunk 5]
```diff
@@ -6,6 +6,8 @@
 
 
 class Pokemon:
+    """Holds information about Pokemon in our own format"""
```
Comment (biggestcookie) [inline game/pokemon.py]: Our own format could be anything we define it to be, so I think that bit is kind of redundant. Maybe describe how it holds the 'current instance' of the Pokemon, since it only exists for as long as the Pokemon is in the player's party or in battle as an enemy.

File: game/pokemon.py [hunk 6]
```diff
@@ -17,16 +19,21 @@ def __init__(
         name: str,
         dex_number: int,
         types: List[Element],
-        total_health: Optional[int] = None
+        total_health: Optional[int] = None,
     ):
+        """ads"""
```
Comment (biggestcookie) [inline game/pokemon.py]: This doesn't tell me anything

File: game/pokemon.py [hunk 7]
```diff
@@ -17,16 +19,21 @@ def __init__(
...
         self.types = types
-        self.total_health = total_health if total_health is not None else random.randint(
-            20, 50)
+        self.total_health = (
+            total_health if total_health is not None else random.randint(20, 50)
+        )
         self.health = self.total_health
 
     def __str__(self) -> str:
+        """asdf"""
```
Comment (biggestcookie) [inline game/pokemon.py]: This also doesn't tell me anything

File: game/typechart.py [hunk 8]
```diff
@@ -26,6 +28,8 @@ def __repr__(self) -> str:
 
 
 class Multiplier(Enum):
+    """Type advantage calulcations"""
```
Comment (biggestcookie) [inline game/typechart.py]: Minor: *`calculations`

File: game/typechart.py [hunk 9]
```diff
@@ -2,6 +2,8 @@
 
 
 class Element(Enum):
+    """Element property declared here"""
```
Comment (biggestcookie) [inline game/typechart.py]: Kind of redundant, I would just say that these are the different options a Pokemon's type can be

File: states/battle.py [hunk 10]
```diff
@@ -2,6 +2,7 @@
 
 
 class Battle:
+    """Will handle the "Battle" function"""
```
Comment (biggestcookie) [inline states/battle.py]: To be specific this is a `state`. It's in the `states` folder and is one of the possible game `states`. Maybe something like 'Game state for player battling' or something and make sure whatever you choose, the other game states' comments are also consistent with this one

File: states/menu.py [hunk 11]
```diff
@@ -5,22 +5,29 @@
 
 
 class MenuOptions(StateOptions):
+    """Variables for Menu options"""
```
Comment (biggestcookie) [inline states/menu.py]: These might be variables, but that's kind of obvious. Specifically, look at how `MenuOptions(StateOptions)` tells you how this class extends off of `StateOptions`. If you look at `StateOptions`, you can see that it is an `enum`. Therefore, `MenuOptions` is also an enum, and probably symbolizes the different options a player can choose in the menu.

File: states/menu.py [hunk 12]
```diff
@@ -5,22 +5,29 @@
...
     EXIT = 3
 
 
 class Menu(State):
+    """Displays options for Menu state change"""
+
     option: Optional[MenuOptions] = None
 
     def __init__(self):
+        """Logic for what occurs if exit option is not choesen"""
```
Comment (biggestcookie) [inline states/menu.py]: You're right, but we might handle the exit option differently in the future. Talk about how this initializes the menu state instead and lists options, then checks options for what a player selects.

Comment (biggestcookie) [reply states/menu.py]: Also minor: `chosen` not `choesen`

File: states/menu.py [hunk 13]
```diff
@@ -5,22 +5,29 @@
...
 class MenuOptions(StateOptions):
+    """Variables for Menu options"""
+
     BATTLE = 1
     TEAM = 2
     EXIT = 3
 
 
 class Menu(State):
+    """Displays options for Menu state change"""
```
Comment (biggestcookie) [inline states/menu.py]: It does a bit more than just display the options, it holds the state of the current menu game state and also lists out options from the player, then reads it from them.

File: states/menu.py [hunk 14]
```diff
@@ -5,22 +5,29 @@
...
     def __init__(self):
+        """Logic for what occurs if exit option is not choesen"""
         super().__init__()
         while self.option is not MenuOptions.EXIT:
             print(TEXT["MAIN"]["ENTRY"])
             self.option = self.check_input(MenuOptions)
             self.choose_option()
 
     def choose_option(self):
+        """Runs respection option chosen"""
```
Comment (biggestcookie) [inline states/menu.py]: I don't think respection is a word

File: states/state.py [hunk 15]
```diff
@@ -18,6 +18,8 @@ def list_options(cls: Type[T]) -> List[str]:
 
 
 class State:
+    """Function for main state and error catcher"""
```
Comment (biggestcookie) [inline states/state.py]: This class is used in multiple places, but never on its own. Specifically, it's extended off of. So it might be best described as 'Base class for game states' and maybe tack on a bit about what functions it provides.

File: states/team.py [hunk 16]
```diff
@@ -1,5 +1,6 @@
 
 
 class Team:
+    """Function for Team Option"""
```
Comment (biggestcookie) [inline states/team.py]: See comments on `states/battle.py`

File: states/state.py [hunk 17]
```diff
@@ -18,6 +20,8 @@ def list_options(cls: Type[T]) -> List[str]:
...
+
     @staticmethod
     def check_input(option_type: Type[T]) -> T:
```
Comment (biggestcookie) [inline states/state.py:28]: Don't forget to document class methods

File: states/state.py [hunk 18]
```diff
@@ -5,10 +5,12 @@
...
 from config.config import TEXT
 
-T = TypeVar('T')
+T = TypeVar("T")
 
 
 class StateOptions(IntEnum):
+    """Gives Menu Option values"""
+
     def __repr__(self) -> str:
```
Comment (biggestcookie) [inline states/state.py:14]: Don't forget to document class methods

File: game/player.py [hunk 19]
```diff
@@ -5,10 +5,14 @@
...
+
+        if len(self.team) < 1:
             self.team.append(pokeapi.get_random_pokemon_from_api())
```
Comment (biggestcookie) [inline game/player.py:18]: For `PLAYER`: Don't forget to document global variables.

**Review by biggestcookie — CHANGES_REQUESTED** [review]
Please review all and address all comments

File: api/pokeapi.py (hunk 1 above)
Comment (ghost) [reply api/pokeapi.py]: Removed period

File: app.py (hunk 2 above)
Comment (ghost) [reply app.py]: Changed to `Entry point for the application`

File: game/pokemon.py (hunk 6 above)
Comment (ghost) [reply game/pokemon.py]: Changed to `Property of Pokemon initialized`

File: game/pokemon.py (hunk 7 above)
Comment (ghost) [reply game/pokemon.py]: Changed to `Converts the information of Pokemon to string`

File: game/typechart.py (hunk 8 above)
Comment (ghost) [reply game/typechart.py]: Changed to `calculations`

File: game/typechart.py (hunk 9 above)
Comment (ghost) [reply game/typechart.py]: Changed to `Possible types that Pokemon can be`

File: config/config.py (hunk 3 above)
Comment (ghost) [reply config/config.py]: Changed to `Text that is fed to user as they navigate the application`

File: game/player.py (hunk 4 above)
Comment (ghost) [reply game/player.py]: Changed to `Fills "Team" array with a random Pokemon from PokeApi`

File: game/pokemon.py (hunk 5 above)
Comment (ghost) [reply game/pokemon.py]: Changed to `Holds information about Pokemon in current instance of the application`

File: states/battle.py (hunk 10 above)
Comment (ghost) [reply states/battle.py]: Changed to `Game state for player battling`

File: states/menu.py (hunk 12 above)
Comment (ghost) [reply states/menu.py]: Changed to `Initializes the menu state and lists available options checking against user input if exit is not chosen`

File: states/menu.py (hunk 11 above)
Comment (ghost) [reply states/menu.py]: Changed to `Enum values for options available in the menu`

File: states/menu.py (hunk 13 above)
Comment (ghost) [reply states/menu.py]: Changed to `Displays, holds, and reads options for Menu state change`

File: states/menu.py (hunk 14 above)
Comment (ghost) [reply states/menu.py]: This is not a word. Changed to `respective`

File: states/state.py (hunk 15 above)
Comment (ghost) [reply states/state.py]: Changed to `Base class for game states`

File: states/team.py (hunk 16 above)
Comment (ghost) [reply states/team.py]: Changed to `Displays team for player choosing Team state`

File: states/state.py (hunk 17 above)
Comment (ghost) [reply states/state.py:28]: Added description `Returns the the options as a string for application to print to user` to classmethod and `Receives user input runs application accordingly even in the case an option not available is chosen` tp staticmethod

File: states/state.py (hunk 18 above)
Comment (ghost) [reply states/state.py:14]: Check staticmethod comment above

File: game/player.py (hunk 19 above)
Comment (ghost) [reply game/player.py:18]: Added the part that these player variable is saved as a global variable

File: app.py (hunk 2 above)
Comment (biggestcookie) [reply app.py]: You sure about that 
image

File: game/player.py [hunk 20]
```diff
@@ -5,10 +5,14 @@
 
 
 class Player:
+    """Holds current information about the player which is saved as a global variable"""
+
     team: List[Pokemon] = []
 
     def __init__(self):
-        if (len(self.team) < 1):
+        """Fills "Team" array with a random Pokemon from PokeApi and saved as a global variable"""
```
Comment (biggestcookie) [inline game/player.py:13]: Minor: don't capitalize references to variables.
`Fills the player's team with a random Pokemon from PokeApi`
The `Player` class doesn't actually do any of the saving to the global `PLAYER` variable, so I wouldn't include that.

File: game/player.py (hunk 19 above)
Comment (biggestcookie) [reply game/player.py:18]: Minor: don't capitalize references to variables.
`Fills the player's team with a random Pokemon from PokeApi`
The Player class doesn't actually do any of the saving to the global PLAYER variable, so I wouldn't include that.

File: game/pokemon.py [hunk 21]
```diff
@@ -6,6 +6,8 @@
 
 
 class Pokemon:
+    """Holds information about Pokemon in current instance of the application"""
```
Comment (biggestcookie) [inline game/pokemon.py:9]: Minor: since the word 'Pokemon` can refer to one or several Pokemon, consider changing this to refer to 'a Pokemon`

File: game/pokemon.py [hunk 22]
```diff
@@ -17,16 +19,21 @@ def __init__(
         name: str,
         dex_number: int,
         types: List[Element],
-        total_health: Optional[int] = None
+        total_health: Optional[int] = None,
     ):
+        """Property of Pokemon initialized"""
```
Comment (biggestcookie) [inline game/pokemon.py]: It's actually properties and not a single property, but actually I think docs on `__init__()` methods might be kind of redundant in general so you can remove them

File: game/typechart.py [hunk 23]
```diff
@@ -2,6 +2,8 @@
 
 
 class Element(Enum):
+    """Possible types that Pokemon can be"""
```
Comment (biggestcookie) [inline game/typechart.py]: Minor: 'a Pokemon'

File: states/menu.py [hunk 24]
```diff
@@ -5,22 +5,29 @@
...
     EXIT = 3
 
 
 class Menu(State):
+    """Displays, holds, and reads options for Menu state change"""
+
     option: Optional[MenuOptions] = None
 
     def __init__(self):
+        """Initializes the menu state and lists available options checking against user input if exit is not chosen"""
```
Comment (biggestcookie) [inline states/menu.py:21]: I said to get rid of `__init__()` comments, but this one does more than a standard init so I think it's good to keep. Mark as resolved 👍

File: states/menu.py [hunk 25]
```diff
@@ -5,22 +5,29 @@
...
         super().__init__()
         while self.option is not MenuOptions.EXIT:
             print(TEXT["MAIN"]["ENTRY"])
             self.option = self.check_input(MenuOptions)
             self.choose_option()
 
     def choose_option(self):
+        """Runs respective state chosen"""
```
Comment (biggestcookie) [inline states/menu.py]: Consider changing to 'Enters next game state based on player choice'

File: states/state.py (hunk 17 above)
Comment (biggestcookie) [reply states/state.py:28]: `Receives user input runs application accordingly even in the case an option not available is chosen` is a bit long, so consider chopping off the rest of the sentence starting at `even`. It's good documentation but I think the finer details can be discovered if need be.

Comment (biggestcookie) [reply states/state.py:28]: Also, minor: `Returns the the options`

File: states/team.py (hunk 16 above)
Comment (biggestcookie) [reply states/team.py]: Minor: make this more consistent with `states/battle.py`'s docstring:
`"""Game state for player battling"""`

File: states/menu.py [hunk 26]
```diff
@@ -5,22 +5,29 @@
...
 class MenuOptions(StateOptions):
+    """Enum values for options available in the menu"""
+
     BATTLE = 1
     TEAM = 2
     EXIT = 3
 
 
 class Menu(State):
+    """Displays, holds, and reads options for Menu state change"""
```
Comment (biggestcookie) [inline states/menu.py:16]: Minor: make this more consistent with `state/battle.py`'s docstring
`"""Game state for player battling"""`
so maybe go with
`"""Game state for main menu"""`
The `Displays, holds, and reads options for state change` is a good description, so maybe move that to the `State` class in `states/state.py` since essentially all game states will be exhibiting this behavior.

File: game/typechart.py [hunk 27]
```diff
@@ -26,6 +28,8 @@ def __repr__(self) -> str:
 
 
 class Multiplier(Enum):
+    """Type advantage calculations"""
```
Comment (biggestcookie) [inline game/typechart.py]: The calculations aren't actually performed here, so more accurately, this is the possible factors a type can multiplied against. 
`Enum values for possible type advantage factors`

File: game/pokemon.py (hunk 22 above)
Comment (ghost) [reply game/pokemon.py]: Removed this comment

File: game/typechart.py (hunk 23 above)
Comment (ghost) [reply game/typechart.py]: Changed to `Possible types that a Pokemon can be`

File: states/state.py [hunk 28]
```diff
@@ -5,21 +5,28 @@
 
 from config.config import TEXT
 
-T = TypeVar('T')
+T = TypeVar("T")
 
 
 class StateOptions(IntEnum):
+    """Gives Menu Option values"""
```
Comment (biggestcookie) [inline states/state.py]: `"""Base enum class for possible actions in game state"""`

File: states/menu.py (hunk 13 above)
Comment (biggestcookie) [reply states/menu.py]: Minor: make this more consistent with state/battle.py's docstring
`"""Game state for player battling"""`
so maybe go with
`"""Game state for main menu"""`
The Displays, holds, and reads options for state change is a good description, so maybe move that to the State class in states/state.py since essentially all game states will be exhibiting this behavior.

File: states/menu.py (hunk 25 above)
Comment (ghost) [reply states/menu.py]: Considered changing to `Enters next game state based on player choice` and did so

File: config/config.py (hunk 3 above)
Comment (biggestcookie) [reply config/config.py]: That's a bit wordy, but I can't think of a better description

File: states/state.py (hunk 28 above)
Comment (ghost) [reply states/state.py]: Changed to `Base enum class for possible actions in game stat`

File: game/player.py (hunk 4 above)
Comment (biggestcookie) [reply game/player.py]: Minor: don't capitalize references to variables.
`Fills the player's team with a random Pokemon from PokeApi`
The Player class doesn't actually do any of the saving to the global PLAYER variable, so I wouldn't include that.

File: game/typechart.py (hunk 27 above)
Comment (ghost) [reply game/typechart.py]: Changed to `Enum values for possible type advantage factors`

File: game/pokemon.py (hunk 5 above)
Comment (biggestcookie) [reply game/pokemon.py]: Minor: since the word 'Pokemon' can refer to one or several Pokemon, consider changing this to refer to 'a Pokemon'

File: states/state.py (hunk 28 above)
Comment (biggestcookie) [reply states/state.py]: I sure hope it's `game state` and not `game stat`

Comment (ghost) [reply states/state.py]: It's definitely `game state` now

File: game/player.py (hunk 4 above)
Comment (ghost) [reply game/player.py]: Changed to `Fills the player's team with a random Pokemon from PokeApi`

File: game/pokemon.py (hunk 5 above)
Comment (ghost) [reply game/pokemon.py]: Changed to `Holds information about a Pokemon in current instance of the application`

File: states/menu.py (hunk 13 above)
Comment (ghost) [reply states/menu.py]: Changed the description for `State` in `states/state,py` to `Base class for game state. Displays, holds, and reads options for Menu state change` and the description for `Menu` in `state/menu.py` to `Game state for main menu`

File: states/team.py (hunk 16 above)
Comment (ghost) [reply states/team.py]: Changed to `Team state for player viewing team`

File: states/state.py (hunk 17 above)
Comment (ghost) [reply states/state.py:28]: Changed static method description to `Receives user input runs application accordingly` and stopped stuttering in the classmethod description

**Review by ghost — COMMENTED** [review]
Please review together

File: states/state.py [hunk 29]
```diff
@@ -5,21 +5,28 @@
...
 
 class StateOptions(IntEnum):
+    """Base enum class for possible actions in game state"""
+
     def __repr__(self) -> str:
         return f"{self.value} - {self.name}"
 
     @classmethod
     def list_options(cls: Type[T]) -> List[str]:
+        """Returns the options as a string for application to print to user"""
```
Comment (biggestcookie) [inline states/state.py]: Minor: `list of strings`, since the return type is not a single string

Comment (ghost) [reply states/state.py]: Changed to `Returns the options as a list of strings for application to print to user`
